sentinelone/komand_sentinelone/connection/connection.py

```diff
@@ -295,24 +300,33 @@ def get_item_ids_by_hash(self, blacklist_hash: str):
                 ids.append(restriction.get("id"))
             return ids
 
-        raise PluginException(cause="The hash does not exist to unblacklist.", assistance="Please enter a hash that has been blacklisted.")
+        raise PluginException(cause="The hash does not exist to unblacklist.",
+                              assistance="Please enter a hash that has been blacklisted.")
```

Comment:
I'm not sure if we want to fail here. Ultimately the user wants a hash to be removed from a blacklist. If the hash did not exist in a blacklist in the first place, does it matter? Isn't their end goal still accomplished?

Reply:
@mrinehart-r7 Good shout here. Turns out this was a remnant and wasn't being hit, even when a user intends to unblacklist a hash that does not exist in the blacklist. I added a more general exception in case some other kinds of errors are hit.

On a similar note, I also changed some of the logic to blacklisting. In the current version, if a hash is already blacklisted and you try to blacklist again, the S1 API responds with an error saying that the hash is already blacklisted. Now, the action will check to see if it is already blacklisted. If it is already blacklisted, the action will skip that hash and return with a success. Otherwise, it will do the blacklisting. See https://github.com/rapid7/insightconnect-plugins/pull/776/files#diff-dc05c921e89d57f5feb11f3b7618a3be14df8afdcf8e9d28ca8c87661365fb48R271-R314

Reply:
Cool, this will need to be a major version bump since functionality has changed, and we will need to update the S1 workflows too match 👍 Thanks for addressing this!